backends/cxxrtl/cxxrtl_backend.cc

```diff
@@ -1835,6 +1835,8 @@ struct CxxrtlWorker {
 			f << ",\n";
 			inc_indent();
 				for (auto &init : mem->inits) {
+					if (init.removed)
+						continue;
```

Comment:
If the inits are actually overlapping here, this is flat out unsound, right? Can you add an assert to make sure this doesn't happen?

Reply:
Uh, why?

The way this works is:

- after creating the helper, there are no `removed` inits
- `coalesce_inits` is called immediately afterwards, non-overlapping inits are left alone, overlapping inits are marked as removed and new, coalesced, inits are created in their place
- the `dump-memory` function skips over `removed` inits to avoid printing the inits that have now been effectively replaced by the coalesced ones

Reply:
Ah, makes sense.